rfc/rfc-0000-supply-chain-tracing.md

```diff
@@ -0,0 +1,182 @@
+- `.status.resources[*].outputs[*].lastTransitionTime` time of the most recent observed change of value. this is not the last time the value observed. This value is helpful to know if a supply chain is progressing. As this field is the result of observation of an edge triggered change, it should not be relied upon when accuracy matters. This is the same behavior as the lastTransitionTime field within a Condition.
+- `.status.resources[*].observedGeneration` the `.metadata.generation` of the stamped resource the outputs are read from. This value is not read from the `.status.observeredGeneration` field that may exist on the stamped resource. A future RFC may choose to take the stamped resource's observed generation under advisement, as that is a change to how Cartographer processes resources, it is inappropriate for this RFC.
+
+In the future, additional information about particular resources may be added. Like an indication of health with an error message when unhealthy. Support is not included in this RFC as it will require additional support within the template to know how to interpret the health of a resource and should be defined in a separate RFC.
+
+# Migration
+[migration]: #migration
+
+All of the data collected by this RFC exists within Cartographer today but is not exposed outwardly by the `Workload`. There are no changes to the runtime behavior required. All changes to existing CRDs are net new and will not break existing clients.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Some users may consider the runtime behavior of a `ClusterSupplyChain` to be an implementation detail that a user creating a `Workload` should not have visibility into. That a particular resource is created by Cartographer for a `Workload` may be viewed as an information disclosure. This change shines light on the internal behavior of a supply chain. The exact definition for each template is not exposed, unless the user also has access to view that resource. The outputs produced for a resource are the most sensitive values captured, but should never contain any confidential data.
+
+If there is significant user concern, a toggle could be developed to partially or fully disable tracing. Such a toggle is not in scope for this RFC.
+
+# Alternatives
+[alternatives]: #alternatives
+
+The alternative to adding trace information to the Workload status is to continue not adding that information and require clients to reconstruct the values on their own. Other RFCs listed below under prior art have broader scopes that treat tracing as a side effect of deeper semantic changes to how Cartographer processes Workloads.
+
+# Prior Art
+[prior-art]: #prior-art
+
+Several RFC have attempted to make stronger pushes into this space, including:
+
+- [First draft of RFC 0014](https://github.com/vmware-tanzu/cartographer/pull/274)
+- [Introduce RFC 18 Workload Report Artifact Provenance](https://github.com/vmware-tanzu/cartographer/pull/519)
+
+# Unresolved Questions
```

Comment:
To point out a possible risk:

We have another RFC on tracing that would duplicate much of this information if adopted. The use cases for that RFC remain and I haven't heard progress on alternate work that would need to be done at the level of all of the choreographed components of a supply chain if that RFC were to be rejected. E.g. at the moment I would predict that we're going to move forward with some version of that RFC.

An important note about tracing is that it is artifact centric rather than resource centric. E.g. wherein this proposal reports 1 artifact per resource, there is good reason for the tracing RFC to report more than 1 artifact for each resource.

If we adopt this RFC and later adopt the tracing RFC we'll need to decide:
- if we want to have two longish fields on the workload that have much of the same information, or
- if we want to make a breaking change and alter what's proposed here.

Reply:
Please add a link to the RFC and ideally the parts of the RFC which you feel are relevant